book/src/rfcs/benchmarking.md

```diff
@@ -0,0 +1,266 @@
+the kimchi level more precisely.
+
+### Pickles time
+
+This is similar to snarky time, mostly differing in the overhead of the
+recursion. At this level beyond some regression we will probably have mostly
+application circuits being benchmarked.
+
+### Pickles memory
+
+Similar to snarky, this will probably be mostly regression tests and maybe some
+example application to give users an idea of what to expect.
+
+## Benchmarks
+
+Here is a non exhaustive list of benchmarks.
+
+### Kimchi hash chain
+
+Benchmarking different aspects of a circuit proving a hash chain.
+
+- The length of the hash chain can easily be a parameter allowing to measure
+different lengths.
+- We want memory and time for compilation, proving and verification, it will be
+implemented as several benchmarks but they will share most code.
+- The number of constraints can be easily shown for each length.  
+- This benchmark with some specific length can be used as regression tests in CI.
+- The full benchmark with several lengths can be shown at discretion of those
+interested running locally, or even added as optional to CI with some way to
+trigger it.
+
+### Pickles hash chain
+
+Same as kimchi hash chain but in pickles, we have less to cover here because we
+already have the kimchi ones, some time and memory benchmarks could be used for
+regressions and to observe the overhead, but having parameters like different
+circuits sizes may be redundant.  A new  possible benchmark would be to have an
+alternative circuit that uses recursion to make the hash chain instead of a
+single circuit. It could be interesting to have a comparison to the non
+recursive approach.
+
+### General kimchi benchmarks
+
+Some general benchmarks covering compilation, proving and verifying, both time
+and memory. Ideally all of them will be parameterized with different circuit
+sizes and a subset can be used as regression tests. We could also add here
+benchmarks for the different operations of our polynomial commitments scheme.
+
+# Drawbacks
+
+Beyond the possibility of spending the time in any other project the main
+drawback I would be the cost, both in CI running time and dollars. This will
+probably be a tradeoff between how much to spend and how much to run in CI.
+
+# Rationale and alternatives
+
+The alternative would be to handle benchmarks in a case by case basis, it may be
+cheaper at short term to not have to design and implement this shared
+infrastructure. Ultimately would be a matter of how much work will be saved
+working on benchmarks. The suggested design may also incentivize the creation of
+more benchmarks by lowering to barriers to create them.
+
+# Prior art
+
+Right now there are just a few benchmarks covering specific cases that can be
+run for command line, mostly in the proof-systems repo. Almost all benchmarks
+are also limited to measuring only time.
+
+# Unresolved questions
+
+- While the tools for rust are mostly covered, specific tools for ocaml have yet
```

Comment:
Do you intend on sorting out the ocaml parts before starting the project? If not, let’s remove them all from the RFC and make a new RFC later with the ocaml parts.